src/Splat.Autofac/AutofacDependencyResolver.cs

```diff
@@ -65,183 +107,90 @@ public bool HasRegistration(Type serviceType, string contract = null)
-                var registrationIndex = 0;
-                while (registrationIndex < registrationCount)
-                {
-                    var componentRegistration = registrations[registrationIndex];
-
-                    var isCandidateForRemoval = GetWhetherServiceRegistrationMatchesSearch(
-                        componentRegistration.Services,
-                        serviceType,
-                        contract);
-                    if (isCandidateForRemoval)
-                    {
-                        registrations.RemoveAt(registrationIndex);
-                        candidatesForRemoval.Add(componentRegistration);
-                        registrationCount--;
-                    }
-                    else
-                    {
-                        registrationIndex++;
-                    }
-                }
-
-                if (candidatesForRemoval.Count == 0)
-                {
-                    // nothing to remove
-                    return;
-                }
-
-                if (candidatesForRemoval.Count > 1)
-                {
-                    // need to re-add some registrations
-                    var reAdd = candidatesForRemoval.Take(candidatesForRemoval.Count - 1);
-                    registrations.AddRange(reAdd);
-
-                    /*
-                    // check for multi service registration
-                    // in future might want to just remove a single service from a component
-                    // rather than do the whole component.
-                    var lastCandidate = candidatesForRemoval.Last();
-                    var lastCandidateRegisteredServices = lastCandidate.Services.ToArray();
-                    if (lastCandidateRegisteredServices.Length > 1)
-                    {
-                        //
-                        // builder.RegisterType<CallLogger>()
-                        //    .AsSelf()
-                        //    .As<ILogger>()
-                        //    .As<ICallInterceptor>();
-                        var survivingServices = lastCandidateRegisteredServices.Where(s => s.GetType() != serviceType);
-                        var newRegistration = new ComponentRegistration(
-                            lastCandidate.Id,
-                            lastCandidate.Activator,
-                            lastCandidate.Lifetime,
-                            lastCandidate.Sharing,
-                            lastCandidate.Ownership,
-                            survivingServices,
-                            lastCandidate.Metadata);
-                        registrations.Add(newRegistration);
-                    }
-                    */
-                }
-
-                RemoveAndRebuild(registrations);
-            }
-        }
+        [Obsolete("Because Autofac 5+ containers are immutable, UnregisterCurrent method is not available anymore.")]
```

Comment:
This obsolete tag is visible only if you're working directly with the AutofacDependencyResolver, and it neatly crosses out the methods in Visual Studio. That's why I thought it could be useful.
If you still think it's better to not have it, I will remove it.

Reply:
Leave them in is my opinion due to the VS feature and automatic warnings at compile time.

Reply:
VS warnings were my reasoning for [Obsolete] as well.